drivers/soundwire/generic_bandwidth_allocation.c

```diff
@@ -260,7 +262,7 @@ static int sdw_compute_port_params(struct sdw_bus *bus)
 
 	ret = sdw_get_group_count(bus, &group);
 	if (ret < 0)
-		goto out;
+		return ret;
```

Comment:
actually maybe you can keep the goto out here? that way there is a single kfree for group->rates in all cases?

Reply:
No, sdw_get_group_count() could be failed to allocate group->rates and we can't free group->rates in that case. See line 229
```
	group->rates = kcalloc(group->max_size, sizeof(int), GFP_KERNEL);
	if (!group->rates)
		return -ENOMEM;
```

Reply:
ok